sematic/ui/src/loginGoogle.tsx

```diff
@@ -0,0 +1,77 @@
+import Alert from "@mui/material/Alert/Alert";
+import Paper from "@mui/material/Paper/Paper";
+import { CredentialResponse, GoogleLogin, GoogleOAuthProvider } from "@react-oauth/google";
+import { useAtom } from "jotai";
+import { useCallback, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import logo from "./Fox.png";
+import { useAppContext, userAtom } from "./hooks/appHooks";
+import { useHttpClient } from "./hooks/httpHooks";
+import { GoogleLoginPayload } from "./Payloads";
+
+
+export default function GoogleLoginPage() {
+  const [error, setError] = useState<Error | undefined>(undefined);
+  const { googleAuthClientId } = useAppContext()
+
+  const [, setUser] = useAtom(userAtom);
+
+  const navigate = useNavigate();
+
+  const { fetch } = useHttpClient();
+
+  const onGoogleLoginSuccess = useCallback(
+    async (credentialResponse: CredentialResponse) => {
+      try {
+        const payload: GoogleLoginPayload = await fetch({
+          url: "/login/google",
+          method: "POST",
+          body: {
+            token: credentialResponse.credential,
+          },
+        });
+        setError(undefined);
+        setUser(payload.user);
+        navigate(-1);
+      } catch (error) {
+        setError(error as Error);
+      }
+    }, [setError, setUser, navigate, fetch]);
+
+  return <Paper
+    sx={{
+      width: 200,
+      p: 5,
+      textAlign: "center",
+      top: "50%",
+      left: "50%",
+      position: "absolute",
+      transform: "translateY(-50%) translateX(-50%)",
+    }}
+    variant="outlined"
+  >
+    <img
+      src={logo}
+      width="50px"
+      alt="Sematic logo"
+      style={{ marginBottom: "30px" }}
+    />
+
+    {error ? <Alert severity="error">{error.message}</Alert> : <></>}
+
+    {!error &&
+      <GoogleOAuthProvider
+        clientId={googleAuthClientId!}
+      >
+        <GoogleLogin
+          text="signin_with"
+          logo_alignment="center"
+          onSuccess={onGoogleLoginSuccess}
+          onError={() => {
+            setError(Error("Unauthorized user"));
```

Comment:
Also print a debug log with the actual response from Google?

Reply:
I tried to do that but found the `onError` callback is not passed with any arguments. So there might be no easy way to emit a log. (There might be hard ways though, but might not worth the effort)